tests/base.py

```diff
@@ -0,0 +1,377 @@
+                    self.assertTrue(field_selected, msg="Field not selected.")
+            else:
+                # Verify only automatic fields are selected
+                expected_automatic_fields = self.expected_automatic_fields().get(cat['tap_stream_id'])
+                selected_fields = self.get_selected_fields_from_metadata(catalog_entry['metadata'])
+                self.assertEqual(expected_automatic_fields, selected_fields)
+
+    @staticmethod
+    def get_selected_fields_from_metadata(metadata):
+        selected_fields = set()
+        for field in metadata:
+            is_field_metadata = len(field['breadcrumb']) > 1
+            # if field['metadata'].get('inclusion') is None and is_field_metadata:  # BUG_SRCE-4313 remove when addressed
+            #     print("Error {} has no inclusion key in metadata".format(field))  # BUG_SRCE-4313 remove when addressed
+            #     continue  # BUG_SRCE-4313 remove when addressed
+            inclusion_automatic_or_selected = (
+                field['metadata']['selected'] is True or \
+                field['metadata']['inclusion'] == 'automatic'
+            )
+            if is_field_metadata and inclusion_automatic_or_selected:
+                selected_fields.add(field['breadcrumb'][1])
+        return selected_fields
+
+
+    @staticmethod
+    def select_all_streams_and_fields(conn_id, catalogs, select_all_fields: bool = True):
+        """Select all streams and all fields within streams"""
+        for catalog in catalogs:
+            schema = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])
+
+            non_selected_properties = []
+            if not select_all_fields:
+                # get a list of all properties so that none are selected
+                non_selected_properties = schema.get('annotated-schema', {}).get(
+                    'properties', {}).keys()
+
+            connections.select_catalog_and_fields_via_metadata(
+                conn_id, catalog, schema, [], non_selected_properties)
+
+    # def set_replication_methods(self, conn_id, catalogs, replication_methods):
+
+    #     replication_keys = self.expected_replication_keys()
+
+    #     for catalog in catalogs:
+
+    #         replication_method = replication_methods.get(catalog['stream_name'])
+
+    #         if replication_method == self.INCREMENTAL:
+    #             replication_key = list(replication_keys.get(catalog['stream_name']))[0]
+    #             replication_md = [{ "breadcrumb": [], "metadata": {'replication-key': replication_key, "replication-method" : replication_method, "selected" : True}}]
+    #         else:
+    #             replication_md = [{ "breadcrumb": [], "metadata": {'replication-key': None, "replication-method" : "FULL_TABLE", "selected" : True}}]
+
+    #         connections.set_non_discoverable_metadata(
+    #             conn_id, catalog, menagerie.get_annotated_schema(conn_id, catalog['stream_id']), replication_md)
```

Comment:
Just double checking we want this commented out

Reply:
Yeah, I can't tell based on the UI/docs if we are able to select different replication methods or not. If we aren't it won't be needed.

Reply:
Removed.